File: src/_03_BehavioralPattern/_03_08_State/java/after/PublicState.java
package _03_BehavioralPattern._03_08_State.java.after;

public class PublicState implements State {
  
  private OnlineCourse onlineCourse;

  public PublicState(OnlineCourse onlineCourse) {
    this.onlineCourse = onlineCourse;
  }

  @Override
  public void addStudent(Student student) {
    this.onlineCourse.getStudents().add(student);
  }

  @Override
  public void addReview(Student student, String review) {
    this.onlineCourse.getReviews().add(review);
  }

  @Override
  public String toString() {
    return "<Public 상태>";
  }
  
}

File: src/_03_BehavioralPattern/_03_08_State/PublicState.js
import State from "./State.js";

/**
 * @typedef { import("./OnlineCourse").default } OnlineCourse
 * @typedef { import("./Student").default } Student
 */

export default class PublicState extends State {
  /** @type { OnlineCourse } */
  _onlineCourse;

  /** @param { OnlineCourse } onlineCourse */
  constructor(onlineCourse) {
    super();
    this._onlineCourse = onlineCourse;
  }

  /** 
   * @override
   * @param { Student }  
   */
  addStudent(student) {
    this._onlineCourse.getStudents().push(student);
  }

  /**
   * @override
   * @param { Student } _student 
   * @param { string } review 
   */
  addReview(_student, review) {
    this._onlineCourse.getReviews().push(review);
  }
}